### PR TITLE
Custom check tests

### PR DESCRIPTION
**Pull request by HEskandari**

### CODE REVIEW

File: src/ServiceControl.PersistenceTests/PersistenceTestCollection.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    public class PersistenceTestCollection : IEnumerable
+    {
+        public IEnumerator GetEnumerator()
+        {
+            yield return new PersistenceDataStoreFixture(DataStoreType.InMemory);
+            yield return new PersistenceDataStoreFixture(DataStoreType.RavenDb);
+            yield return new PersistenceDataStoreFixture(DataStoreType.SqlDb);
+        }
+    }
+}
```
Comment (jpalac) [inline src/ServiceControl.PersistenceTests/PersistenceTestCollection.cs:15]: This is where we need that check for the existence of the sqlconnection in env variables

File: src/ServiceControl/CustomChecks/CustomChecksComponent.cs [hunk 2]
```diff
@@ -34,7 +34,7 @@ public override void Configure(Settings settings, IHostBuilder hostBuilder)
                         serviceCollection.AddSingleton<ICustomChecksDataStore, RavenDbCustomCheckDataStore>();
                         break;
                     case DataStoreType.SqlDb:
-                        serviceCollection.AddSingleton<ICustomChecksDataStore>(sp => new SqlDbCustomCheckDataStore(settings.SqlStorageConnectionString, sp.GetService<IDomainEvents>()));
+                        serviceCollection.AddSingleton<ICustomChecksDataStore>(sp => new SqlDbCustomCheckDataStore(settings.SqlStorageConnectionString));
```
Comment (jpalac) [inline src/ServiceControl/CustomChecks/CustomChecksComponent.cs:37]: I saw that the `DomainEvents` wasn't required and have also removed it from this code in my [PR](https://github.com/Particular/ServiceControl/pull/2933) - perhaps we should try and merge these 2 together

**Review by tmasternak — APPROVED** [review]

File: src/ServiceControl.PersistenceTests/PersistenceTestCollection.cs (hunk 1 above)
Comment (HEskandari) [reply src/ServiceControl.PersistenceTests/PersistenceTestCollection.cs:20]: Updated

File: src/ServiceControl/CustomChecks/CustomChecksComponent.cs (hunk 2 above)
Comment (HEskandari) [reply src/ServiceControl/CustomChecks/CustomChecksComponent.cs]: Rebasing with the branch solved this.

File: src/ServiceControl.Persistence.InMemory/InMemoryCustomCheckDataStore.cs [hunk 3]
```diff
@@ -10,8 +11,47 @@ namespace ServiceControl.Persistence.InMemory
...
+        public Task<CheckStateChange> UpdateCustomCheckStatus(CustomCheckDetail detail)
+        {
+            if (storage.ContainsKey(detail.CustomCheckId))
+            {
+                var storedCheck = storage[detail.CustomCheckId];
+                if (storedCheck.HasFailed == detail.HasFailed)
+                {
+                    return Task.FromResult(CheckStateChange.Unchanged);
+                }
+            }
```
Comment (mikeminutillo) [inline src/ServiceControl.Persistence.InMemory/InMemoryCustomCheckDataStore.cs:25]: Should this update the state here?

**Review by mikeminutillo — APPROVED** [review]

File: src/ServiceControl.Persistence.InMemory/InMemoryCustomCheckDataStore.cs (hunk 3 above)
Comment (HEskandari) [reply src/ServiceControl.Persistence.InMemory/InMemoryCustomCheckDataStore.cs:25]: There is a bit of code that maps from `CustomCheckDetail` that does this exact thing in other persistences. That potentially can be refactored into something reusable and removed from individual persistence layers. I'll update that in a separate PR.

**Pull request merged**